client/components/d3-base/index.js

```diff
@@ -117,7 +121,8 @@ export default class D3Base extends Component {
 	}
 
 	updateParams = () => {
-		this.setState( { params: this.state.getParams( this.chartRef.current ) } );
+		const params = this.state.getParams( this.chartRef.current );
+		this.setState( { params, updateParams: false } );
```

Comment:
This behavior of this component is so difficult to grok. It's not clear to me why we need all those changes. Could you clarify that for me?